From: Ifran Ali <khanu.i.i.k.2000@gmail.com>
Subject: [PATCH] cli: do not repeat the previous command on an empty line

An empty line no longer re-runs the previous command in either the hush
or the simple parser, so a stray Enter on the serial console cannot
repeat a flash erase or write.

Signed-off-by: Ifran Ali <khanu.i.i.k.2000@gmail.com>
---
diff --git a/common/cli_hush.c b/common/cli_hush.c
index fe8fe93b..a6ad8a10 100644
--- a/common/cli_hush.c
+++ b/common/cli_hush.c
@@ -1044,17 +1044,14 @@ static void get_user_input(struct in_str *i)
 	clear_ctrlc();
 	do_repeat = 0;
 	if (i->promptmode == 1) {
-		if (console_buffer[0] == '\n'&& flag_repeat == 0) {
-			strcpy(the_command,console_buffer);
+		if (console_buffer[0] == '\n') {
+			strcpy(the_command, console_buffer);
+			flag_repeat = 0;
+			do_repeat = 0;
 		}
 		else {
-			if (console_buffer[0] != '\n') {
-				strcpy(the_command,console_buffer);
-				flag_repeat = 1;
-			}
-			else {
-				do_repeat = 1;
-			}
+			strcpy(the_command, console_buffer);
+			flag_repeat = 1;
 		}
 		i->p = the_command;
 	}
diff --git a/common/cli_simple.c b/common/cli_simple.c
index 266c4443..2ad17901 100644
--- a/common/cli_simple.c
+++ b/common/cli_simple.c
@@ -287,7 +287,7 @@ void cli_simple_loop(void)
 			strlcpy(lastcommand, console_buffer,
 				CONFIG_SYS_CBSIZE + 1);
 		else if (len == 0)
-			flag |= CMD_FLAG_REPEAT;
+			continue;
 #ifdef CONFIG_BOOT_RETRY_TIME
 		else if (len == -2) {
 			/* -2 means timed out, retry autoboot
